### ISSUE TITLE
No output in jupyter notebook -suddenly

### ISSUE THREAD

**Issue by johann-petrak** [holoviz/hvplot#463]
Using python 3.6 on Linux with 
* holoviews 1.13.2
* hvplot 0.5.2
* bokeh 2.0.2
* IPython 7.1.1

When I do something simple like:
```
df.hvplot(kind="scatter")
```
A red "Out[nn]"  is shown and nothing else. This frequently happens in notebooks where the same plot showed fine earlier. At some point it stopped working. 

Under my imports:
```
import holoviews as hv
hv.extension("bokeh")
import hvplot
import hvplot.pandas
```
The following messages are shown:
```
Javascript error adding output!
TypeError: Bokeh.set_log_level is not a function
See your browser Javascript console for more details
```

On the Chrome javascript console there are many warnings and the following errors:
```
Uncaught TypeError: Cannot read property 'embed_items_notebook' of undefined 
    at embed_document (<anonymous>:5:20)
    at <anonymous>:8:5
    at <anonymous>:24:3
    at eval (eval at append_javascript (outputarea.js:763), <anonymous>:97:28)
    at Array.forEach (<anonymous>)
    at window._Events.handle_add_output (eval at append_javascript (outputarea.js:763), <anonymous>:86:13)
    at window._Events.dispatch (jquery.min.js:5237)
    at window._Events.elemData.handle (jquery.min.js:5044)
    at Object.trigger (jquery.min.js:8471)
    at window._Events.<anonymous> (jquery.min.js:8549)
events.js:33 Exception in event handler for output_added.OutputArea TypeError: Cannot use 'in' operator to search for '1754' in undefined
    at window._Events.handle_add_output (eval at append_javascript (outputarea.js:763), <anonymous>:103:45)
    at window._Events.dispatch (jquery.min.js:5237)
    at window._Events.elemData.handle (jquery.min.js:5044)
    at Object.trigger (jquery.min.js:8471)
    at window._Events.<anonymous> (jquery.min.js:8549)
    at Function.each (jquery.min.js:367)
    at jQuery.fn.init.each (jquery.min.js:202)
    at jQuery.fn.init.trigger (jquery.min.js:8548)
    at jQuery.fn.init.events.trigger (events.js:31)
    at OutputArea.append_output (outputarea.js:355) Arguments(2)
```

In the Firefox console I see the following errors:
```
Comm promise not found for comm id hv-extension-comm comm.js:127
    comm_msg comm.js:127
    proxy jQuery
    _handle_iopub_message kernel.js:1224
    _finish_ws_message kernel.js:1016
    _msg_queue kernel.js:1007
TypeError: root.Bokeh.embed is undefinedplotting-libs.ipynb line 97 > scriptElement:5:3
    embed_document plotting-libs.ipynb line 97 > scriptElement:5
    <anonymous> plotting-libs.ipynb line 97 > scriptElement:8
    <anonymous> plotting-libs.ipynb line 97 > scriptElement:24
    handle_add_output main.min.js:97
    forEach self-hosted:225
    handle_add_output main.min.js:86
    jQuery 7
    trigger events.js:31
    append_output outputarea.js:355
    fromJSON outputarea.js:1046
    fromJSON codecell.js:564
    fromJSON notebook.js:2660
    load_notebook_success notebook.js:3146
    proxy jQuery
Exception in event handler for output_added.OutputArea TypeError: "right-hand side of 'in' should be an object, got undefined"
    handle_add_output main.min.js:103
    jQuery 7
    trigger events.js:31
    append_output outputarea.js:355
    fromJSON outputarea.js:1046
    fromJSON codecell.js:564
    fromJSON notebook.js:2660
    load_notebook_success notebook.js:3146
    proxy jQuery
```

What I find frustrating is that the same notebook worked fine a bit earlier and there is nothing
obvious that has been changed to explain why it suddenly does not work and no helpful error message as to what might be going wrong.

**Comment by johann-petrak** [comment]
When I create a completely new notebook (in the same Firefox browser) and copy over the same cells I used in the original notebook where things stopped to work, everything works fine: the error messages under the cell with the imports are not shown, and no error messages appear on the Javascript console and the graph is shown correctly. 

Going back to the original notebook with exactly the same cells and running those cells produces the same errors again. Even after restarting the Kernel. 

It feels like something "broke"  the notbook itself and changed it so that things do not work properly and there seems to be no way to reset things to the state when it still did work.

**Comment by johann-petrak** [comment]
OK some more research shows that the presence of other cells in the notebook, even though those cells do not get executed after jupyter notebook or the kernel gets restarted, seems to cause this. 
I created a copy and removed all the other cells and then, after restarting the notebook, things work. 
The notebook compares a number of Python plotting libs and contains graphs from matplotlib, seaborn, plotly and pygal. 

However just using the Jupyter notebook command to clear all the outputs, then restarting the kernel and reloading the notebook page in the browser still does not change anything: the hvplot method still does not show anything in that notebook.

**Comment by johann-petrak** [comment]
OK, one more attempt: turns out that clearing the output with the shortcut from within the notebook does not seem to remove it from the actual notebook file.

Tried to clean the output by creating a copy like so:
```
jupyter nbconvert original.ipynb --to notebook --ClearOutputPreprocessor.enabled=True --stdout > cleaned.ipynb
```
Running the cells in the cleaned.ipynb with a new kernel still does not make it work.

**Comment by philippjfr** [comment]
Thanks for the detailed writeup here. Do you, by any chance, still have a copy of the notebook in its broken state?

**Comment by johann-petrak** [comment]
This is a (gzip-compressed, github does not allow .ipynb extension) notebook where running the cells marked with `# RUN 1` and `# RUN 2` (at the beginning) followed by `# RUN 3` and `# Run 4` (close to the end) does not work for me on two different computers. 
[plotting-libs.ipynb.gz](https://github.com/holoviz/hvplot/files/4639335/plotting-libs.ipynb.gz)

**Comment by johann-petrak** [comment]
@philippjfr  did the notebook allow you to replicate the problem?

**Comment by philippjfr** [comment]
Sorry I've been totally swamped, hoping to get to it later in the week.

**Comment by SarahOst** [comment]
I had the same issue with my own notebook browser independent.

I am using python 3.7 on Windows 10 with

- holoviews 1.13.3
- hvplot 0.6.0
- bokeh 2.1.1
- IPython 7.16.1.

I discovered my notebook was working again if I changed markdown cells with Bokeh as text/headline, e.g. "## Bokeh" or "### Bokeh". Error occurs if the Bokeh-Markdown-cell exist at any position in the notebook (before/after import or plot).

As soon as you change **all** cells with "# Bokeh" to something like "# aBokeh" you shouldn't have the error anymore.

I tested this workaround successfully with my own and with @johann-petrak 's notebook.

**Comment by johann-petrak** [comment]
OK, this is beyond weird: when I open my notebook, first change the Markdown cell to not contain "# Bokeh" as suggested by @SarahOst , then run the cells marked with "RUN 1" to "RUN 4" the plot indeed gets shown.

**Comment by SarahOst** [comment]
I can even specify a tiny bit more:

- even if you run the notebook with the failure causing "# Bokeh" you can still change the markdown cell and rerun code cells and they will work now (so, there it is not necessarily to change it first to make the notebook work again)

- failure occurs only with the capital version "# Bokeh" and not with "# bokeh"

**Comment by mattpap** [comment]
ref https://github.com/bokeh/bokeh/issues/10002#issuecomment-654810980

**Comment by philippjfr** [comment]
Oh wow thanks @mattpap. That's truly awful! 😢

**Comment by jlstevens** [comment]
The PR @mattpap references above has now been merged so hopefully this will be fixed in the next bokeh release.

**Comment by LinuxIsCool** [comment]
Having this issue now suddenly. 

bokeh==3.2.2
holoviews==1.17.1
hvplot==0.8.4
jupyterlab==3.6.3
jupyterlab-pygments==0.2.2
jupyterlab-unfold==0.2.2
jupyterlab_server==2.24.0
panel==1.2.1
param==1.13.0

Python 3.10.12



Actually first, what was happening was that axis were displayed, but no line charts or heatmaps could be displayed. Area charts still plotted. I then updated libraries:

  • Updating bokeh (3.2.1 -> 3.2.2)
  • Updating holoviews (1.17.0 -> 1.17.1)
  
And now I'm not getting any output at all in jupyter, not even blank axis.

**Comment by philippjfr** [comment]
>Actually first, what was happening was that axis were displayed, but no line charts or heatmaps could be displayed

This was related to the Bokeh webgl renderer breaking in certain cases, which should have been fixed by bokeh 3.2.2. Do you see any errors on your JS console?
